xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/chaining/BlockStateChainingListener.java

```diff
@@ -278,6 +278,10 @@ public void beginDocument(MetaData metadata)
     public void beginDefinitionDescription()
     {
         ++this.inlineDepth;
+        if (this.definitionListDepth.isEmpty()) {
```

Comment:
Why not encapsulating the content of the method in an `if (!this.definitionListDepth.isEmpty()) {` instead of adding artificially an element? Would that break something else?

Reply:
Hm, I thought so, but now I'm not so sure anymore. It seems that [`getListItemIndex`](https://github.com/xwiki/xwiki-rendering/blob/51bd06ccfb9d47d697158270642c84f8e7531a1b/xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/chaining/BlockStateChainingListener.java#L193-L196) already checks if the stack is non-empty, so this might actually be a better solution... It will break the printing of the list items in both XWiki syntax and plain text, but it might be cleaner.

Reply:
And +1 for having a warning in the else branch then.

Reply:
I've adjusted the code. I've decided not to log a warning as it is hard to provide relevant information for admins to perform any actions, in particular without a stack trace. For now, I'm just logging a warning. A possibility could also be to make this a warning when deprecation logging is enabled as in https://github.com/xwiki/xwiki-platform/blob/05d353688490b416a90fcf9ba116cca04e8a9b06/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java#L2434-L2437 but to me that doesn't really seem fitting, what do you think?

Reply:
> to me that doesn't really seem fitting

Yeah, does not feel like a deprecation. In the case of cache document modification, I used it because modifying directly the cached document used to be kind of officially OK.